### PR TITLE
Moved dropped event to raise after dropped item is moved

### PR DESCRIPTION
**Pull request by riccardi48**
## About the PR

Currently the DroppedEvent is raised before the entity's location is set to the target location. This moves it after it is moved and rotated.
## Why / Balance

The dropped event is used for dropping things like beakers so they can break if you slip. Use cases that use the location that an item has been dropped are currently difficult to implement as you can't get that location without some very funky workarounds.
There shouldn't be any cases where this would break anything, and if you need the location of the user which dropped the item, the user entity is sent in the event.
## Technical details

Moved the code which moves the dropped item into `DoDrop()`
Added `targetDropLocation` input to `DoDrop()`, uses the same logic as before.
Moved `DroppedEvent` to be raised after the rotation of the grid is applied to the entity. Same reasoning as above.
## Test plan

If there isn't proper integration test coverage for this then there might be an edge case which turns up at some point. Otherwise not much testing is needed.
## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have tested this pull request and written instructions on how to test it
- [X] I have added media to this PR or it does not require an in-game showcase.

## Breaking changes

`DroppedEvent` is now raised after the dropped entity has been moved and rotated to the grid rotation. If you require the old location, get the transform of the user entity passed into the event.

### CODE REVIEW

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs [hunk 1]
```diff
@@ -255,6 +247,15 @@ public virtual void DoDrop(Entity<HandsComponent?> ent,
...
         }
 
+        if (targetDropLocation != null)
+        {
+            var (itemPos, itemRot) = TransformSystem.GetWorldPositionRotation(entity.Value);
+            // otherwise, also move dropped item and rotate it properly according to grid/map
+            var origin = new MapCoordinates(itemPos, Transform(entity.Value).MapID);
+            var target = TransformSystem.ToMapCoordinates(targetDropLocation.Value);
+            TransformSystem.SetWorldPositionRotation(entity.Value, GetFinalDropCoordinates(ent, origin, target, entity.Value), itemRot);
+        }
```
Comment (metalgearsloth) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs:257]: Long-term would be better to not be doing world stuff.
